src/core/include/hydrogen/core_action_controller.h

```diff
@@ -98,9 +98,11 @@ class CoreActionController : public H2Core::Object {
 		 *
 		 * \param songPath Absolute path to the .h2song file to be
 		 *    opened.
+		 * \param bRestartDriver Whether or not to restart the audio
+		 *    driver after successfully loading the new song.
 		 * \return true on success
 		 */
-		bool openSong( const QString& songPath );
+		bool openSong( const QString& songPath, bool bRestartDriver );
```

Comment:
This "restart drivers after song load" logic seems to be kind of strange to me and makes the song loading process even more complicated. But i guess there no easy way around, at least i also don't see a better solution at the moment :confused:

Reply:
Yeah. I added another commit in order to streamline the loading a little more. But yes. It's rather complicated and this adds another layer. I removed this argument and replaced it by some checks in the `CoreActionController` itself. Only if Hydrogen is under session management the audio driver will be restarted. 

Still, the most puzzling (and probably redundant) part for me is that the Song has to be loaded by the GUI. This leads to really awkward program flow as the NsmClient reads a song from disk and stores it into a variable. The GUI than takes this song and asks the `Hydrogen` class to set it. But when just omitting the GUI part, the application freezes. I haven't looked into it much but this core -> gui -> core part can most probably be simplified. (But this is not directly related to NSM so I did not attempted it in this PR).